test/qa/test_connect6.py

```diff
@@ -12,99 +12,81 @@
     login,
     network,
 )
+from test_connect import disconnect_base_test
 
 
-def setup_module(module):  # noqa: ARG001
+def setup_function(function):  # noqa: ARG001
     daemon.start()
     login.login_as("default")
+    logging.log()
+    print(sh.nordvpn.set.ipv6.on())
+
 
+def teardown_function(function):  # noqa: ARG001
+    logging.log(data=info.collect())
+    logging.log()
 
-def teardown_module(module):  # noqa: ARG001
+    print(sh.nordvpn.set.ipv6.off())
     sh.nordvpn.logout("--persist-token")
+    sh.nordvpn.set.defaults()
     daemon.stop()
 
 
-def setup_function(function):  # noqa: ARG001
-    logging.log()
+def connect_base_test(group: str = (), name: str = "", hostname: str = "", ipv6 = True):
+    """
+    Connects to a NordVPN server and performs a series of checks to ensure the connection is successful.
 
+    Parameters:
+    group (str): The specific server name or group name to connect to. Default is an empty string.
+    name (str): Used to verify the connection message. Default is an empty string.
+    hostname (str): Used to verify the connection message. Default is an empty string.
+    ipv6 (bool): If True, checks if IPv6 connection is available. Default is True.
+    """
 
-def teardown_function(function):  # noqa: ARG001
-    logging.log(data=info.collect())
-    logging.log()
+    output = sh.nordvpn.connect(group, _tty_out=False)
+    print(output)
 
+    assert lib.is_connect_successful(output, name, hostname)
+    assert network.is_connected()
 
-@pytest.mark.parametrize(("tech", "proto", "obfuscated"), lib.TECHNOLOGIES_WITH_IPV6[:-1])
+    if ipv6:
+        assert network.is_ipv6_connected(20)
```

Comment:
why so many retries `20`? If something goes wrong it would take at least 20 seconds to fail.

Reply:
Good point, I will check this out